### PR TITLE
[CONTINT-3862] Inject container name in DD_ENTITY_ID with admission controller

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR modifies the admission controller to inject the container name in DD_ENTITY_ID with the admission controller. 

### Motivation
Retrieve container tags even when container-id can't be found.

### Describe how to test/QA your changes
Deploy the agent on Kubernetes with the admission controller enabled and `DD_ADMISSION_CONTROLLER_INJECT_CONFIG_INJECT_CONTAINER_NAME=true`.
Deploy a dummy application with multiple containers and multiple init containers. Make sure an env var `DD_ENTITY_ID=en-(init.)${POD_UID}/${CONTAINER_NAME}` is injected in each container/init container.
Test [notebook](https://dddev.datadoghq.com/notebook/7908325/test-dogstatsd-with-container-name-injected-by-admission-controller?cell-rb1mqheb-live=false&cell-rb1mqheb-range=301596&cell-rb1mqheb-start=1713185364957).

Test application:
```
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: dogstatsd-udp
  labels:
    app: dogstatsd-udp
    admission.datadoghq.com/enabled: "true"
spec:
  replicas: 1
  selector:
    matchLabels:
      app: dogstatsd-udp
  template:
    metadata:
      labels:
        app: dogstatsd-udp
    spec:
      initContainers:
        - name: app
          image: ubuntu
          command:
            - /bin/bash
            - -c
            - |
              export DEBIAN_FRONTEND=noninteractive
              apt update
              apt install -y netcat bc

              for i in $(seq 1 10); do
                sleep 5
                printf 'initcustom.metric.name:%s|g|#dd.internal.entity_id:%s,foo:bar\n' "$(echo "s($(date +%s)/60)" | bc -l)" "$DD_ENTITY_ID" | nc -u -q 0 -w60 "$DD_AGENT_HOST" 8125
              done
          env:
            - name: DD_AGENT_HOST
              valueFrom:
                fieldRef:
                  fieldPath: status.hostIP
            - name: DD_INTERNAL_POD_UID
              valueFrom:
                fieldRef:
                  fieldPath: metadata.uid
      containers:
        - name: appp
          image: ubuntu
          command:
            - /bin/bash
            - -c
            - |
              export DEBIAN_FRONTEND=noninteractive
              apt update
              apt install -y netcat bc

              while true; do
                while true; do
                  printf 'custom.metric.name:%s|g|#dd.internal.entity_id:%s,foo:bar\n' "$(echo "s($(date +%s)/60)" | bc -l)" "$DD_ENTITY_ID" 
                  sleep 10
                done | nc -u -q0 -w60 "$DD_AGENT_HOST" 8125
              done
          env:
            - name: DD_AGENT_HOST
              valueFrom:
                fieldRef:
                  fieldPath: status.hostIP
```

### CODE REVIEW

**Review by estherk15 — APPROVED** [review]

**Review by louis-cqrl — APPROVED** [review]
LGTM for file ASC owns

File: pkg/clusteragent/admission/mutate/config/config.go [hunk 1]
```diff
@@ -201,6 +217,56 @@ func injectionMode(pod *corev1.Pod, globalMode string) string {
...
+	return true
+}
+
+// injectFullIdentity injects `DD_INTERNAL_CONTAINER_NAME`, `DD_INTERNAL_POD_UID` and `DD_ENTITY_ID`
+// as `en-(init.)$(DD_INTERNAL_POD_UID)/$(DD_INTERNAL_CONTAINER_NAME)`.
+func injectFullIdentity(pod *corev1.Pod) bool {
+	injected := false
+	podStr := common.PodString(pod)
+	for i := range pod.Spec.Containers {
+		injected = injectIdentityInContainer(&pod.Spec.Containers[i], "", podStr)
```
Comment (adel121) [inline pkg/clusteragent/admission/mutate/config/config.go:262]: Should this be `injected = injectIdentityInContainer(&pod.Spec.Containers[i], "", podStr) || injected`?

Comment (AliDatadog) [reply pkg/clusteragent/admission/mutate/config/config.go:262]: Good catch 👍

**Review by adel121 — CHANGES_REQUESTED** [review]
left a comment

File: pkg/clusteragent/admission/mutate/config/config.go [hunk 2]
```diff
@@ -201,6 +217,56 @@ func injectionMode(pod *corev1.Pod, globalMode string) string {
...
+		_ = log.Errorf("Cannot inject identity into nil container")
+		return false
+	}
+	// Do not override DD_ENTITY_ID if it's already set
+	if lo.ContainsBy(container.Env, func(v corev1.EnvVar) bool { return v.Name == ddEntityIDEnvVarName }) {
+		log.Debugf("Ignoring container '%s' in pod %s: env var '%s' already exist", container.Name, podStr, ddEntityIDEnvVarName)
+		return false
+	}
+
+	// We can and should override DD_INTERNAL_* variables if they are already set
```
[the remark this answers is not in the document]
Comment (AliDatadog) [reply pkg/clusteragent/admission/mutate/config/config.go:231]: IMO this is acceptable. It would mean that it was set manually by the client.

**Review by adel121 — APPROVED** [review]

**Pull request merged**